### PR TITLE
Downside protection summary and loading phase step

### PR DESCRIPTION
**Pull request by piekczyk**
# [Downside protection summary and loading phase step](https://app.shortcut.com/oazo-apps/story/3610/downside-protection-completed)

<please insert a clubhouse link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>
- added downside protection complete summary and loading phase step
  
## How to test 🧪
  <Please explain how to test your changes>
- heroku / locally
    
## Definition of done ✔️

- [ ] Acceptance criteria for each issue met
- [ ] Unit tests written where needed and passing
- [ ] End-to-end tests for happy path
- [ ] Documentation updated <When applicable>
- [ ] Project builds without errors
- [ ] Non-functional requirements met
- [ ] Code reviewed and functionality tested by the reviewer (locally, Heroku, etc.)
- [ ] Feature verified and accepted by Product Owner
- [ ] Project deployed to production environment

### CODE REVIEW

File: components/dumb/TxStatusSection.tsx [hunk 1]
```diff
@@ -1,26 +1,32 @@
...
 
 export interface TxStatusSectionProps<A extends TxMeta> {
   txState: TxState<A> | undefined
 }
 
 export function TxStatusSection<A extends TxMeta>(props: TxStatusSectionProps<A>) {
   const { t } = useTranslation()
+
+  const txStatus = props.txState?.status
+  const txHash = (props.txState as any)?.txHash
```
Comment (rkrasiuk) [inline components/dumb/TxStatusSection.tsx:16]: do you have to cast to `any` here?

**Review by rkrasiuk — DISMISSED** [review]

File: components/dumb/TxStatusSection.tsx (hunk 1 above)
Comment (piekczyk) [reply components/dumb/TxStatusSection.tsx:16]: yep, TxState type doesn't have txHash definition.

Comment (rkrasiuk) [reply components/dumb/TxStatusSection.tsx:17]: is it worth adding it there?

Comment (piekczyk) [reply components/dumb/TxStatusSection.tsx:17]: We take this type from @oasisdex/transactions library so if we would like to add this type it has to be done there. This casting regarding txStatus is currently common in our repo. I see that guys from our team contribute there so for now I would suggest to take a note of that and rise this topic on planning. I believe that after some investigation maybe more type definitions could be improved there so we could treat it as separate story.

**Review by rkrasiuk — APPROVED** [review]

**Pull request merged**